### PR TITLE
Upgrade error_prone to 2.18.0 (from 2.16)

### PR DESCRIPTION
**Pull request by carterkozak**
==COMMIT_MSG==
Upgrade error_prone to 2.18.0 (from 2.16)
==COMMIT_MSG==

## Possible downsides?
New checks added in https://github.com/google/error-prone/releases/tag/v2.17.0 and https://github.com/google/error-prone/releases/tag/v2.18.0 may prevent upgrades from auto-merging.

### CODE REVIEW

**Comment by a-k-g** [comment]
Stumbled upon this as we we're hitting https://github.com/google/error-prone/issues/2321 on an internal repo. The fix is in 2.17.0 but not sure if that is any easier to merge.

File: baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/LogsafeArgName.java [hunk 1]
```diff
@@ -54,6 +55,7 @@ public LogsafeArgName() {
         this.unsafeParamNames = ImmutableSet.of();
     }
 
+    @Inject
     public LogsafeArgName(ErrorProneFlags flags) {
```
Comment (carterkozak) [inline baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/LogsafeArgName.java:59]: Caught by [`InjectOnBugCheckers`](https://errorprone.info/bugpattern/InjectOnBugCheckers) in [2.18.0](https://github.com/google/error-prone/releases/tag/v2.18.0).

File: build.gradle [hunk 2]
```diff
@@ -52,6 +52,9 @@ allprojects {
                 check("PreferSafeLogger", CheckSeverity.OFF)
                 check("PreferSafeLoggingPreconditions", CheckSeverity.OFF)
                 check("PreconditionsConstantMessage", CheckSeverity.OFF)
+                // temporarily opt out of error-prone in order to upgrade checkerframework.
+                // This should be removed as soon as baseline is next upgraded within this repo.
+                enabled = false
```
Comment (carterkozak) [inline build.gradle:57]: I've compiled this using a snapshot of itself, things seem good, the problem is upgrading dependencies makes our existing baseline classes (applied to this project) incompatible with the upgraded checkerframework.

File: gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineErrorProne.java [hunk 3]
```diff
@@ -175,7 +175,9 @@ private static void configureErrorProneOptions(
                 "PreferImmutableStreamExCollections",
                 "UnusedVariable",
                 // See VarUsage: The var keyword results in illegible code in most cases and should not be used.
-                "Varifier");
+                "Varifier",
+                // Yoda style should not block baseline upgrades.
+                "YodaCondition");
```
Comment (schlosna) [inline gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineErrorProne.java:180]: There are a few new 🌶️ checkers, I would bet on [AvoidObjectArrays](https://errorprone.info/bugpattern/AvoidObjectArrays), [Finalize](https://errorprone.info/bugpattern/Finalize), [YodaCondition](https://errorprone.info/bugpattern/YodaCondition) likely need some manual intervention/suppression upon rollout (though in general I think we want those checks enabled)

### 2.17:

[AvoidObjectArrays](https://errorprone.info/bugpattern/AvoidObjectArrays)
[Finalize](https://errorprone.info/bugpattern/Finalize)
[IgnoredPureGetter](https://errorprone.info/bugpattern/IgnoredPureGetter)
[ImpossibleNullComparison](https://errorprone.info/bugpattern/ProtoFieldNullComparison)
[MathAbsoluteNegative](https://errorprone.info/bugpattern/MathAbsoluteNegative)
[NewFileSystem](https://errorprone.info/bugpattern/NewFileSystem)
[StatementSwitchToExpressionSwitch](https://errorprone.info/bugpattern/StatementSwitchToExpressionSwitch)
[UnqualifiedYield](https://errorprone.info/bugpattern/UnqualifiedYield)

### 2.18:

[InjectOnBugCheckers](https://errorprone.info/bugpattern/InjectOnBugCheckers)
[LabelledBreakTarget](https://errorprone.info/bugpattern/LabelledBreakTarget)
[UnusedLabel](https://errorprone.info/bugpattern/UnusedLabel)
[YodaCondition](https://errorprone.info/bugpattern/YodaCondition)

**Review by schlosna — APPROVED** [review]

File: build.gradle (hunk 2 above)
Comment (carterkozak) [reply build.gradle:57]: This flag has been removed with the baseline upgrade here: https://github.com/palantir/gradle-baseline/pull/2503/files#diff-49a96e7eea8a94af862798a45174e6ac43eb4f8b4bd40759b5da63ba31ec3ef7L55-L57

**Pull request merged**